libretroshare/src/retroshare/rsgxsifacehelper.h

```diff
@@ -292,25 +290,55 @@ struct RsGxsIfaceHelper
 	 * Useful for blocking API implementation.
 	 * @param[in] token token associated to the request caller is waiting for
 	 * @param[in] maxWait maximum waiting time in milliseconds
+	 * @param[in] checkEvery time in millisecond between status checks
 	 */
 	RsTokenService::GxsRequestStatus waitToken(
 	        uint32_t token,
-	        std::chrono::milliseconds maxWait = std::chrono::milliseconds(500) )
+	        std::chrono::milliseconds maxWait = std::chrono::milliseconds(500),
+	        std::chrono::milliseconds checkEvery = std::chrono::milliseconds(2))
 	{
+#if defined(__ANDROID__) && (__ANDROID_API__ < 24)
+		auto wkStartime = std::chrono::steady_clock::now();
+		int maxWorkAroundCnt = 10;
+LLwaitTokenBeginLabel:
+#endif
 		auto timeout = std::chrono::steady_clock::now() + maxWait;
 		auto st = requestStatus(token);
 		while( !(st == RsTokenService::FAILED || st >= RsTokenService::COMPLETE)
 		       && std::chrono::steady_clock::now() < timeout )
 		{
-			std::this_thread::sleep_for(std::chrono::milliseconds(2));

```

Comment:
Sleeping for only 2 ms between two checks is probably far too small. You'll be creating lots of CPU loops. I would advise 100 or 200 ms here.

Reply:
100ms sound reasonable for operations like channel creation that take lot of time but it is far too much for simple reads, I have been measuring this value even on old android phones and for reads operations 2ms is more then enough (probably the system does some caching), also this PR doesn't change the default value it just make it configurable, I agree that long operations like channel creation might be optimized by passing a longer value as parameters ;)